### DISCUSSION TITLE
How to rotate each pixel in an EBSD map by a specified angle and save it in a new ebsd variable?

### DISCUSSION THREAD

**Discussion by FloZr** [mtex-toolbox/mtex#2224] [Ask Anything] [answered]
My goal is to substract out the rigid body rotation of an experiment that bends a metal sheet by 180° following the following puplication doi:10.1007/s12613-015-1156-1. 
I do that by calculating the orientation matrix from the euler angles and then rotating this matrix by an angle and calculating back to euler angles. 
The problem arises when i save this new orientation in an ebsd variable, this step seems to take an unreasonable amount of time (for ~5000 points about 45 seconds). I do this with 
```matlab
ebsd.orientations(index)=rotated_euler_angles
```
, with rotated_euler_angles being the rotated euler angles from the rotated orientation matrix:
```matlab
 rotated_euler_angles =inv(orientation.byMatrix(G_transformed))
```
All of this happens in a loop that gets each pixel in a gridified ebsd map.

Is there a better way to save the new orientations that is more efficient??

**Comment by FloZr** [comment, marked as answer]
I solved this by only calculating the angles in a loop and then applying all the rotation at once with rotation.byAxisAngle.

**Comment by ralfHielscher** [comment]
The proper way to perform this operation should be

```matlab
% define the rigid body rotation
rot = rotation.byAxisAngle(xvector,90*degree)

% apply this rotations to the orientation data
ebsd2 = rotate(ebsd,rot,'keepXY')
```
Ralf.

**Reply by FloZr** [reply]
This is more or less what i did, but as each pixel needs a different rotation I applied the rotation inside a loop, this was the reason it was so slow. The better way seems to be calculating the rotations seperately and then applying them all at once with the method you mentioned.

Thank you for the reply!
